docs/user/custom_worker.md

```diff
@@ -0,0 +1,149 @@
+scratch. For example:
+
+```py
+from garage.sampler import DefaultWorker
+
+class MyWorker(DefaultWorker):
+    """Initialize a worker.
+
+    Args:
+        seed(int): The seed to use to initialize random number generators.
+        max_path_length(int or float): The maximum length paths which will
+            be sampled. Can be (floating point) infinity.
+        worker_number(int): The number of the worker where this update is
+            occurring. This argument is used to set a different seed for each
+            worker.
+
+    Attributes:
+        agent(Policy or None): The worker's agent.
+        env(gym.Env or None): The worker's environment.
+
+    """
+    def __init__(
+            self,
+            *,  # Require passing by keyword, since everything's an int.
+            seed,
+            max_path_length,
+            worker_number):
+        ...
+        super().__init__(seed=seed,
+                         max_path_length=max_path_length,
+                         worker_number=worker_number)
+        ...
+
+    def rollout(self):
+        ...
+```
+
+## Custom Worker for RL2
+
+In RL2, different from general RL algorithms, multiple trajectories are sampled
+in one trial/meta batch. Thus we can move the original rollout into a loop:
+
+```py
+from garage.sampler import DefaultWorker
+
+class RL2Worker(DefaultWorker):
+    """Initialize a worker for RL2.
+
+    Args:
+        seed(int): The seed to use to initialize random number generators.
+        max_path_length(int or float): The maximum length paths which will
+            be sampled. Can be (floating point) infinity.
+        worker_number(int): The number of the worker where this update is
+            occurring. This argument is used to set a different seed for each
+            worker.
+        n_paths_per_trial (int): Number of trajectories sampled per trial/
+            meta batch. Policy resets in the beginning of a meta batch,
+            and obtain `n_paths_per_trial` trajectories in one meta batch.
+
+    Attributes:
+        agent(Policy or None): The worker's agent.
+        env(gym.Env or None): The worker's environment.
+
+    """
+    def __init__(
+            self,
+            *,  # Require passing by keyword, since everything's an int.
+            seed,
+            max_path_length,
+            worker_number,
+            n_paths_per_trial=2):
+        self._n_paths_per_trial = n_paths_per_trial
+        super().__init__(seed=seed,
+                         max_path_length=max_path_length,
+                         worker_number=worker_number)
+
+    def rollout(self):
+        for _ in range(self._n_paths_per_trial):
+            self.start_rollout()
+            while not self.step_rollout():
+                pass
+        return self.collect_rollout()
+```
+
+This is not enough, because in RL2 the agent/policy only resets at the beginning
+of a trail/meta batch. So we need to override `start_rollout()`. And we want to
```

Comment:
can you choose a consistent terminology for trial or meta-batch? Please just use the terminology from the original paper. If there is a conflict, you can explain that the original paper uses the term "foo" and in garage that is gener